Explorer/Assets/DCL/InWorldCamera/PhotoDetail/PhotoDetailController.cs

```diff
@@ -0,0 +1,197 @@
+        private readonly ICameraReelScreenshotsStorage cameraReelScreenshotsStorage;
+        private readonly ISystemClipboard systemClipboard;
+        private readonly IDecentralandUrlsSource decentralandUrlsSource;
+        private readonly IWebBrowser webBrowser;
+        private readonly string shareToXMessage;
+
+        private MetadataSidePanelAnimator metadataSidePanelAnimator;
+        private CancellationTokenSource showReelCts = new ();
+
+        private bool metadataPanelIsOpen = true;
+        private bool isClosing;
+        private int currentReelIndex;
+
+        public override CanvasOrdering.SortingLayer Layer => CanvasOrdering.SortingLayer.Popup;
+
+        public PhotoDetailController(ViewFactoryMethod viewFactory,
+            PhotoDetailInfoController photoDetailInfoController,
+            ICameraReelScreenshotsStorage cameraReelScreenshotsStorage,
+            ISystemClipboard systemClipboard,
+            IDecentralandUrlsSource decentralandUrlsSource,
+            IWebBrowser webBrowser,
+            string shareToXMessage)
+            : base(viewFactory)
+        {
+            this.photoDetailInfoController = photoDetailInfoController;
+            this.cameraReelScreenshotsStorage = cameraReelScreenshotsStorage;
+            this.systemClipboard = systemClipboard;
+            this.decentralandUrlsSource = decentralandUrlsSource;
+            this.webBrowser = webBrowser;
+            this.shareToXMessage = shareToXMessage;
+
+            this.photoDetailInfoController.JumpIn += JumpInClicked;
+        }
+
+        private void ShowDeleteModal()
+        {
+            viewInstance!.deleteReelModal.gameObject.SetActive(true);
+            viewInstance.deleteReelModal.DOFade(1f, viewInstance.deleteModalAnimationDuration);
+        }
+
+        private void HideDeleteModal(InputAction.CallbackContext callbackContext = default) =>
+            viewInstance!.deleteReelModal.DOFade(0f, viewInstance.deleteModalAnimationDuration).OnComplete(() => viewInstance.deleteReelModal.gameObject.SetActive(false));
+
+        private void DeletionModalCancelClick(bool waitForAnimation = true)
+        {
+            async UniTaskVoid DelayedHideDeleteModalAsync()
+            {
+                await UniTask.Delay(ANIMATION_DELAY);
+                HideDeleteModal();
+            }
+
+            if (waitForAnimation)
+                DelayedHideDeleteModalAsync().Forget();
+            else
+                HideDeleteModal();
+        }
+
+        private void JumpInClicked() =>
+            isClosing = true;
+
+        private void ToggleInfoSidePanel()
+        {
+            metadataSidePanelAnimator.ToggleSizeMode(toFullScreen: metadataPanelIsOpen, viewInstance.sidePanelAnimationDuration);
+            metadataPanelIsOpen = !metadataPanelIsOpen;
+        }
+
+        protected override void OnViewShow()
+        {
+            viewInstance!.infoButton.onClick.AddListener(ToggleInfoSidePanel);
```

Comment:
Technically, the controller should never know how the view is implemented, whether it uses a button or other control should be transparent. Views should provide just events. In this case, moving the controls to the controller and deleting the view would be the same as it is now. What do you think?

Reply:
I agree that a layer of abstraction between the two parts would be much more suited. The main reason why I went with this approach is that I've seen it everywhere in the codebase and doing something different felt misleading from an overall perspective